tests/lib/fakestore/store/store.go

```diff
@@ -739,49 +740,109 @@ func (s *Store) retrieveAssertion(bs asserts.Backstore, assertType *asserts.Asse
 	return a, err
 }
 
-func (s *Store) assertionsEndpoint(w http.ResponseWriter, req *http.Request) {
-	assertPath := strings.TrimPrefix(req.URL.Path, "/v2/assertions/")
+func (s *Store) retrieveSequenceFormingAssertion(bs asserts.Backstore, assertType *asserts.AssertionType, sequenceKey []string, sequence int) (asserts.Assertion, error) {
+	if sequence > 0 {
```

Comment:
With the current use case for this function, this if block is dead code which made me question the intent.

From documentation `SeqFormingAssertion` "sequence <= 0 we query for the latest sequence",
which seems to suggest the sequence (-1 or 0) does not play a functional role and this function always retrieves the latest sequence. 

Perhaps this function can be simplified and called something like `retrieveLatestSequenceFormingAssertion` without passing `sequence` and using hardcoded -1 for 'fallback.SeqFormingAssertion'? This then relies on the caller to only call when applicable (sequence > 0) which is already the case.

Reply:
Great point, there was no point in adding that case currently as we never hit it. I have simplified the function as suggested.